components/brave_wallet_ui/common/hardware/ledgerjs/ledger-trusted-transport.ts

```diff
@@ -0,0 +1,34 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { LedgerMessagingTransport } from './ledger-messaging-transport'
+import {
+  LedgerCommand,
+  AuthorizationSuccessCommand
+} from './ledger-messages'
+
+// LedgerTrustedMessagingTransport is the messaging transport object
+// for chrome://wallet and chrome://wallet-panel. It makes calls to the Ledger
+// libraries isolated in chrome-untrusted://ledger-bridge via postMessage,
+// and defines a handler to be run when the untrusted instance is granted access
+// to the user's the Ledger device
+export class LedgerTrustedMessagingTransport extends LedgerMessagingTransport {
+  private onAuthorized?: () => void
+
+  constructor (targetWindow: Window, targetUrl: string, onAuthorized?: () => void) {
+    super(targetWindow, targetUrl)
+    this.onAuthorized = onAuthorized
+    this.addCommandHandler(LedgerCommand.AuthorizationSuccess, this.handleAuthorizationSuccess)
```

Comment:
You're thinking that handleAuthorizationSuccess handler should be defined (and added) in the one step in the constructor?  I can understand the reasoning that if the function isn't called elsewhere, it shouldn't be defined as a method.  In this case though, I'd advocate for keeping as is for code readability.  Whatever we do here, we should also do to the handlers defined on LedgerUntrustedMessagingTransport, since the same principle applies.